lua/neo-tree/sources/filesystem/init.lua

```diff
@@ -34,10 +34,10 @@ end
 local follow_internal = function(callback, force_show, async)
   log.trace("follow called")
   if vim.bo.filetype == "neo-tree" or vim.bo.filetype == "neo-tree-popup" then
-    return
+    return false
   end
   local path_to_reveal = manager.get_path_to_reveal()
-  if not utils.truthy(path_to_reveal) then
+  if not path_to_reveal or not utils.truthy(path_to_reveal) then
```

Comment:
Shouldn't `utils.truthy` cover whatever is being handled by `not path_to_reveal`?

Reply:
Indeed, but it shows diagnostic error so I simply added it. And theoretically it's faster because now we might skip one function call.

Reply:
Well, it's a bit more difficult than that.

```lua
local x = function_return_0_or_nil()
-- type(x) == integer or nil
if not utils.truthy(x) then
  return
end
-- type(x) == integer or nil (still thinks x can be nil)
local y = x + 1 -- RED ERROR

--------
local x = function_return_0_or_nil()
-- type(x) == integer or nil
if not x or not utils.truthy(x) then
  return
end
-- type(x) == integer (x is always integer here)
local y = x + 1 -- HAPPY LSP ;)
```

Reply:
I don't have any issue from diagnostics with this code:
```lua
  local path_to_reveal = manager.get_path_to_reveal()
  if not utils.truthy(path_to_reveal) then
```

`path_to_reveal` is `string|nil`, `utils.truthy` accepts `any`. What error do you get?